From 6c5b3b0ce2cfd716d30b113bb1a9af3ff4edad00 Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@openadk.org>
Date: Wed, 20 Mar 2024 09:45:58 +0100
Subject: [PATCH] riscv64: enable CTORS, fixes C++ applications, found by
 sorear

---
 extra/Configs/Config.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extra/Configs/Config.in b/extra/Configs/Config.in
index 4c3e8d627..7d213e46e 100644
--- a/extra/Configs/Config.in
+++ b/extra/Configs/Config.in
@@ -544,7 +544,7 @@ config LDSO_LD_LIBRARY_PATH
 
 config UCLIBC_CTOR_DTOR
 	bool
-	default y if !TARGET_riscv64 && !TARGET_lm32
+	default y
 	help
 	  If you wish to build uClibc with support for global constructor
 	  (ctor) and global destructor (dtor) support, then answer Y here.
-- 
2.25.1

